src/os/bluestore/BlueStore.cc

```diff
@@ -70,8 +71,30 @@
 #define dout_context cct
 #define dout_subsys ceph_subsys_bluestore
 
-using bid_t = decltype(BlueStore::Blob::id);
+#ifdef WITH_EXPERIMENTAL
+#  define BS_FLAVOR "bluestore-exp"
```

Comment:
Just curious, have you considered doing this with a separate `BlueStore_exp.cc` file?  Depending on how invasive and long-lived experimental bits would be, we can end up with quite an `ifdef` mess.  The patches for promoting some but not all experimental bits to "stable" would be very non-obvious because it would all be `ifdef` wrangling rather than a "here is new code" kind of diff.

A related question: would any code submission against the stable version be expected to make the corresponding changes to the experimental version?  If the experimental code would be allowed to lag behind until it's periodically "rebased", a separate source file makes even more sense IMO.